agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/PerformanceCounterInitializer.java

```diff
@@ -109,23 +111,79 @@ private static void loadCustomJmxPerfCounters(List<Configuration.JmxMetric> jmxX
         collection.add(new JmxAttributeData(jmxElement.name, jmxElement.attribute));
       }
 
-      // Register each entry in the performance container
-      for (Map.Entry<String, Collection<JmxAttributeData>> entry : data.entrySet()) {
+      createMeterPerAttribute(data);
+
+    } catch (RuntimeException e) {
+      try (MDC.MDCCloseable ignored = CUSTOM_JMX_METRIC_ERROR.makeActive()) {
+        logger.error("Failed to register JMX performance counters: '{}'", e.toString());
+      }
+    }
+  }
+
+  // Create a meter for each attribute & declare the callback that reports the metric in the meter.
+  private static void createMeterPerAttribute(
+      Map<String, Collection<JmxAttributeData>> objectAndAttributesMap) {
+    for (Map.Entry<String, Collection<JmxAttributeData>> entry :
+        objectAndAttributesMap.entrySet()) {
+      String objectName = entry.getKey();
+      for (JmxAttributeData jmxAttributeData : entry.getValue()) {
+        GlobalOpenTelemetry.meterBuilder("com.microsoft.applicationinsights.jmx")
+            .setSchemaUrl("app_insights_" + jmxAttributeData.metricName)
+            .build()
+            .gaugeBuilder(jmxAttributeData.metricName.replaceAll(" ", "_"))
```

Comment:
probably worth replacing all other characters that aren't supported by OTel instrument names too